### ISSUE TITLE
Is adding gcc and firends to library path for Vulkan really necessary?

### ISSUE THREAD

**Issue by exzombie** [guibou/nixGL#178]
Not having (yet) gone to test a hudred different programs, I'm wondering what is the purpose of adding stuff like zlib and gcc to `LD_LIBRARY_PATH`. Does Vulkan really not work without them? I tried removing everything added by `nixGLVulkanIntel`, keeping only paths added by `nixGLIntel`, and running vulkan-info, mpv and chromium; all three reported Vulkan was functional.

The reason I ask is that, after going to nixpkgs 24.05, a lot of stuff is broken. The glibc version is newer than on the host OS, so I get glibc errors when a wrapped program tries to start a host-provided executable. To avoid this, one can pin the version of nixpkgs used for nixGL to 22.11, so that the wrappers are built with an older glibc. It seems sensible to use drivers from the same era as the host OS. However, because the Vulkan wrapper pulls is a bunch of other libraries, I now get GLIBCXX version conflicts.

In other words, I'm stuck between a rock and a hard place, and the stuff added to the library path by the vulkan wrapper stands in the way of me getting out of that place and finishing https://github.com/Smona/home-manager/pull/1.

**Comment by karl3wm** [comment]
```
$ nixVulkanIntel vulkaninfo
vulkaninfo: /nix/store/k2a429wpxgfwp4jaacl9iaqw4kxqjaxa-gcc-11.3.0-lib/lib/libstdc++.so.6: version `GLIBCXX_3.4.32' not found (required by vulkaninfo)
vulkaninfo: /nix/store/k2a429wpxgfwp4jaacl9iaqw4kxqjaxa-gcc-11.3.0-lib/lib/libstdc++.so.6: version `CXXABI_1.3.15' not found (required by vulkaninfo)
```
```
$ nixVulkanIntel bash -c "$(nixGLIntel bash -c set | grep LD_LIBRARY_PATH) vulkaninfo"                                      WARNING: lavapipe is not a conformant vulkan implementation, testing use only.
==========
VULKANINFO
==========
...
```
I had to follow OP to make it work.
